src/world/Explosion.php

```diff
@@ -212,13 +245,28 @@ public function explodeB() : bool{
 				if(($t = $this->world->getTileAt($pos->x, $pos->y, $pos->z)) !== null){
 					$t->onBlockDestroyed(); //needed to create drops for inventories
 				}
-				$this->world->setBlockAt($pos->x, $pos->y, $pos->z, $airBlock);
```

Comment:
You should still set the block inside this condition. Fire can't be placed on the TNT block pos in vanilla